nautobot/extras/forms/forms.py

```diff
@@ -997,10 +1005,21 @@ def clean(self):
             for field_name in JOB_OVERRIDABLE_FIELDS:
                 if not cleaned_data.get(f"{field_name}_override", False):
                     cleaned_data[field_name] = getattr(job_class, field_name)
-            if not cleaned_data.get("job_queues_override", False):
-                cleaned_data["job_queues"] = JobQueue.objects.filter(
-                    name__in=getattr(job_class, "task_queues", []) or [settings.CELERY_TASK_DEFAULT_QUEUE]
+            # Get default Job Queue first
+            if not cleaned_data.get("default_job_queue_override", False):
+                meta_task_queues = getattr(job_class, "task_queues", []) or [settings.CELERY_TASK_DEFAULT_QUEUE]
+                cleaned_data["default_job_queue"], _ = JobQueue.objects.get_or_create(
+                    name=meta_task_queues[0], defaults={"queue_type": JobQueueTypeChoices.TYPE_CELERY}
                 )
```

Comment:
Maybe just a nitpick but `form.clean()` ideally shouldn't be creating a DB record, this should be happening in `form.save()`. Not sure how important this is though.